drivers/dma/dw-axi-dmac/dw-axi-dmac-platform.c

```diff
@@ -261,6 +261,15 @@ static u32 axi_chan_get_xfer_width(struct axi_dma_chan *chan, dma_addr_t src,
 	return __ffs(src | dst | len | BIT(max_width));
 }
 
+static u32 axi_dma_encode_msize(u32 max_burst)
+{
+	if (max_burst == 0)
+		return DWAXIDMAC_BURST_TRANS_LEN_1;
+	if (max_burst > 1024)
+		return DWAXIDMAC_BURST_TRANS_LEN_1024;
+	return fls(max_burst) - 2;
```

Comment:
This will underflow for a max_burst of 1 - not typically set by drivers (using 0 to denote "no bursts please" instead) but will fix.